tests/sentry/hybrid_cloud/test_organizationmembermapping.py

```diff
@@ -13,6 +13,22 @@
 
 @control_silo_test(stable=True)
 class OrganizationMappingTest(TransactionTestCase, HybridCloudTestMixin):
+    def test_upsert_stale_user_id(self):
+        assert (
+            organizationmember_mapping_service.upsert_mapping(
+                organization_id=self.organization.id,
+                organizationmember_id=111111,
+                mapping=RpcOrganizationMemberMappingUpdate(
+                    role=self.organization.default_role,
+                    user_id=10001,
+                    email=None,
+                    inviter_id=None,
+                    invite_status=None,
+                ),
+            )
+            is None
```

Comment:
This might be silly, but should we add an assertion here that none of the stale user data in the DB has been modified as well?

Reply:
Sure

Reply:
Ah, so actually, looking at this again -- the whole reason this test needs to exist *is because* there is a constraint between OrganizationMemberMapping and User.  In fact, adding the test that nothing was insert into the db is making the point; there is nothing wrong with there being an entry in the database *except for the fact that the constraint would normally cause an exception and prevent the upsert from working*.

Technically if we removed the constraint and the test still passed, everything would still work.  Does that make sense?

The important parts of this test are that
1.  Inserting stale users should not throw an exception and
2.  It should not return a mapping object, signaling that the last given inputs are stale.

Whether a row is actually inserted is not necessarily important.